test/parallel_exec.jl

```diff
@@ -11,6 +11,90 @@ elseif Base.JLOptions().code_coverage == 2
 end
 addprocs(3; exeflags=`$cov_flag $inline_flag --check-bounds=yes --depwarn=error`)
 
+# Test remote()
+
+let
+    pool = Base.default_worker_pool()
+
+    count = 0
+    count_condition = Condition()
+
+    function remote_wait(c)
+        @async begin
+            count += 1
+            remote(take!)(c)
+            count -= 1
+            notify(count_condition)
+        end
+        yield()
+    end
+
+    w = workers()
+    @test length(w) == 3
+    c1 = RemoteChannel()
+    c2 = RemoteChannel()
+    c3 = RemoteChannel()
+
+    t1 = now()
+
+    @test isready(pool) == true
+    @test count == 0
+    remote_wait(c1)
+    @test count == 1
+    put!(c1, "foo")
+    wait(count_condition)
+    @test count == 0
+
+
+    @test isready(pool) == true
+    @test count == 0
+    remote_wait(c1)
+    @test count == 1
+    remote_wait(c2)
+    @test count == 2
+    remote_wait(c3)
+    @test count == 3
+    @test isready(pool) == false
```

Comment:
May be worthwhile to make these and below tests work with any number of workers by re-writing like 
```
testchannels = [RemoteChannel() for i in 1:nworkers()]
testcount = 0
@test isready(pool) == true
for c in testchannels
    @test count == testcount
     remote_wait(c)
     testcount += 1 
end
@test isready(pool) == false
```

This way if the number of workers changes in the future we do not have to revisit this block of tests. Also more readable.